src/plugins/embeddable/public/lib/panel/embeddable_panel.tsx

```diff
@@ -257,12 +257,21 @@ export class EmbeddablePanel extends React.Component<Props, State> {
   public componentDidMount() {
     if (this.embeddableRoot.current) {
       this.subscription.add(
-        this.props.embeddable.getOutput$().subscribe((output: EmbeddableOutput) => {
-          this.setState({
-            error: output.error,
-            loading: output.loading,
-          });
-        })
+        this.props.embeddable.getOutput$().subscribe(
+          (output: EmbeddableOutput) => {
+            this.setState({
+              error: output.error,
+              loading: output.loading,
+            });
+          },
+          (error) => {
+            this.props.embeddable.destroy();
```

Comment:
The approach LGTM,

One weirdness I noticed: 

1. Load a "broken" dashboard in view move. See Lens embeddable with error 
2. Go to edit mode and try to delete it. There is no option to "DELETE" an embeddable. O_O. 
I think it is important that it still possible to delete an embeddable in this scenario.


Probably we shouldn't call destroy in this scenario, because embeddable still needs to receive update from parent (`viewMode` in this case) that will be used during context menu rendering? 

Probably we could rely on destroy being called when this component unmounts?

Reply:
This is a really good catch. I avoided destroying the embeddable, and added clauses in the `isCompatible` methods of a few problematic actions like Clone, add to library, and unlink from library. Now you should be able to properly remove or replace errorEmbeddables.